ably/ably-ui
# [MAW-244] Add search widget to meganav

Jira Ticket Link / Motivation
----



https://ably.atlassian.net/browse/MAW-244

Add a search widget to the Meganav - see https://www.figma.com/file/S01cEFu4DCQtWABIIYLqMF/Search-in-Meganav?node-id=0%3A1 for details.

Summary of changes
----



See commits for details.

How do you manually test this?
----



On the review app, head to the Meganav component (`/components/meganav`) and test the search widget:
- panel should open only on click
- other panels should not open on hover when search panel is open
- input should focus on open
- when you start typing, a suggestions dropdown should show
- the suggestions dropdown should be navigable with up, down arrows and enter/space for selecting
- on mobile, after you start typing, a clear button should show on the right in the input to remove anything entered in it

Note also that you can add query params to see the component in different variants:
- adding `framework=vw` will display the Ruby version of the Meganav (by default it's React)
- adding `signed-in=true` will display the the signed in version of the Meganav

So for example, `/components/meganav?framework=vw&signed-in=true` will show you the signed in, Ruby version of the Meganav.

The search widget should display identically for all of these.

Reviewer Tasks (optional)
----




Merge/Deploy Checklist
----




Frontend Checklist
----
